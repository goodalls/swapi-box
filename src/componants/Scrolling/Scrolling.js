import React from 'react';
import './Scrolling.css';
import PropTypes from 'prop-types';

const Scrolling = ({ title, episodeId, openingCrawl, releaseDate }) => {

  return (
    <div className="scrolling">
      <div className="fade" />

      <section className="star-wars">
        <div className="crawl">
          <div className="title">
            <p>Episode {episodeId}</p>
            <h1>{title}</h1>
          </div>
          <p>{openingCrawl}</p>

          <div className="title">
            <p>released on: {releaseDate}</p>
          </div>
        </div>
      </section>
    </div>
  );
};

Scrolling.propTypes = {
  title: PropTypes.string,
  episodeId: PropTypes.number,
  openingCrawl: PropTypes.string,
  releaseDate: PropTypes.string
};

export default Scrolling;
