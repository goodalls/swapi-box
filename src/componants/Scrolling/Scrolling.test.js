import React from 'react';
import Scrolling from './Scrolling';
import { shallow } from 'enzyme';

describe('Scrolling', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Scrolling />);
    expect(wrapper).toMatchSnapshot();
  });
});
