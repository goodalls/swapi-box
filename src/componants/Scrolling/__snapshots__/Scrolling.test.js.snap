// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrolling should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scrolling />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <p>
                                        Episode 
                              </p>
                              <h1 />
                    </div>
                    <p />
                    <div
                              className="title"
                    >
                              <p>
                                        released on: 
                              </p>
                    </div>
          </div>
</section>,
      ],
      "className": "scrolling",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <p>
                                    Episode 
                        </p>
                        <h1 />
            </div>
            <p />
            <div
                        className="title"
            >
                        <p>
                                    released on: 
                        </p>
            </div>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <p>
                                Episode 
                </p>
                <h1 />
</div>,
              <p />,
              <div
                className="title"
>
                <p>
                                released on: 
                </p>
</div>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode 
</p>,
                  <h1 />,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  released on: 
</p>,
                "className": "title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "released on: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "released on: ",
                  undefined,
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <section
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <p>
                                                Episode 
                                    </p>
                                    <h1 />
                        </div>
                        <p />
                        <div
                                    className="title"
                        >
                                    <p>
                                                released on: 
                                    </p>
                        </div>
            </div>
</section>,
        ],
        "className": "scrolling",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <p>
                                          Episode 
                            </p>
                            <h1 />
              </div>
              <p />
              <div
                            className="title"
              >
                            <p>
                                          released on: 
                            </p>
              </div>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <p>
                                    Episode 
                  </p>
                  <h1 />
</div>,
                <p />,
                <div
                  className="title"
>
                  <p>
                                    released on: 
                  </p>
</div>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Episode 
</p>,
                    <h1 />,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Episode ",
                        undefined,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Episode ",
                      undefined,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    released on: 
</p>,
                  "className": "title",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "released on: ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "released on: ",
                    undefined,
                  ],
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
